File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int cnt=1;
        int n=nums.size();
        int ele=nums[0];
        for(int i=1;i<n;i++)
        {
            if(cnt<=0)
            {
                cnt=1;
                ele=nums[i];
            }
            else if(ele==nums[i]) cnt++;
            else cnt--;
        }
        //check
        int cnt1=0;

        for(int i=0;i<n;i++)
        {
            if(ele==nums[i]) cnt1++;
        }

        if(cnt1>=(n/2)) return ele;
        return -1;

    }
};

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        c = None 
        n = len(nums)
        for i in range(n):
            if count == 0:
                c = nums[i]
                count = 1
            elif nums[i] == c:
                count+= 1 
            else:
                count-=1
        return c 
       